### PR TITLE
CWAT-52905562: Added friend routes

### PR DESCRIPTION
**Pull request by abhinit75**
Added the following Routes:

1. `GET /friends/`: Retrieves all friends (both source and destination matching id in param)
2. `GET /friends/requests`: Retrieves all pending friends requests from source id, where destination_id is matching id
3.  `POST /friends/`: Creates a friend request
4. `PUT /friends/approve-request`: Updates friend request where source_friend_id, destination_friend_id and is_approved flag it passed
5. `DELETE /friends/:id`:  Removes friendship record based on id of the record in friend table passed

Other notes:
- Added is_approved flag to the friends table to handle pending requests.
- Friendship is mutual - if either friend unfriends, the record is removed.

### CODE REVIEW

File: server/src/api/FriendRoutes.ts
Comment (Kggupta) [inline server/src/api/FriendRoutes.ts:86]: why arent we using the req.body.user.id here?

File: server/src/api/FriendRoutes.ts
Comment (Kggupta) [inline server/src/api/FriendRoutes.ts:128]: this never runs, delete doesnt throw an error if nothing is deleted. also we can fail silently if the friendship doesnt exist imo.

**Review by Kggupta — CHANGES_REQUESTED** [review]

**Comment by Kggupta** [comment]
Why are there so many migrations? i think some of them are overlapping with exisitng migrations in main

**Comment by mary1afshar** [comment]
Looks good, just one question. To send a friend request, I was thinking of doing it by email. Not sure if thats how its done here but it looks like it is done by useID. Is that correct?

File: server/src/api/FriendRoutes.ts
Comment (abhinit75) [reply server/src/api/FriendRoutes.ts:86]: changed to body.user.id

Comment (Kggupta) [reply server/src/api/FriendRoutes.ts:86]: also rename this endpoint to be more informative. like /approve-request

File: server/src/api/FriendRoutes.ts
Comment (Kggupta) [inline server/src/api/FriendRoutes.ts:14]: why was this switched to be from params id?

File: server/src/api/FriendRoutes.ts
Comment (Kggupta) [inline server/src/api/FriendRoutes.ts:41]: also why is this from params? you have the user in all of these requests because the token is authenticated

Comment (abhinit75) [reply server/src/api/FriendRoutes.ts:41]: resolved to body.user.id for all

File: server/src/api/FriendRoutes.ts
Comment (abhinit75) [reply server/src/api/FriendRoutes.ts:14]: resolved to body.user.id for all

File: server/src/api/FriendRoutes.ts
Comment (abhinit75) [reply server/src/api/FriendRoutes.ts:86]: done

**Comment by Kggupta** [comment]
lgtm, i'll let maryam approve once you update the create friendship to use email instead of user id.

File: server/src/api/FriendRoutes.ts
Comment (Kggupta) [inline server/src/api/FriendRoutes.ts:100]: it would probably be better to use user id in this case, because when this endpoint is run we'll have access to the user id at this point.

Comment (abhinit75) [reply server/src/api/FriendRoutes.ts:100]: Fair I was thinking the same

**Review by Kggupta — APPROVED** [review]

**Review by mary1afshar — APPROVED** [review]
LGTM!

**Pull request merged**
